File: src/main/java/architecture/goldenboughs_lib/core/capability/entity/EntityLcLevel.java
package architecture.goldenboughs_lib.core.capability.entity;

import architecture.goldenboughs_lib.api.LcLevel;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

public enum EntityLcLevel implements IEntityLcLevel {
	NULL() {
		@Override
		public @Nullable LcLevel getLcLevel() {
			return null;
		}
	},
	ZAYIN() {
		@Override
		public @NotNull LcLevel getLcLevel() {
			return LcLevel.ZAYIN;
		}
	},
	TETH() {
		@Override
		public @NotNull LcLevel getLcLevel() {
			return LcLevel.TETH;
		}
	},
	HE() {
		@Override
		public @NotNull LcLevel getLcLevel() {
			return LcLevel.HE;
		}
	},
	WAW() {
		@Override
		public @NotNull LcLevel getLcLevel() {
			return LcLevel.WAW;
		}
	},
	ALEPH() {
		@Override
		public @NotNull LcLevel getLcLevel() {
			return LcLevel.ALEPH;
		}
	}
}

File: src/main/kotlin/architecture/goldenboughs_lib/api/capability/entity/EntityLcLevel.kt
package architecture.goldenboughs_lib.api.capability.entity

import architecture.goldenboughs_lib.api.LcLevel

/**
 * 一些预制对象
 */
enum class EntityLcLevel : IEntityLcLevel {
	NULL {
		override fun getLcLevel(): LcLevel? {
			return null
		}
	},
	ZAYIN {
		override fun getLcLevel(): LcLevel {
			return LcLevel.ZAYIN
		}
	},
	TETH {
		override fun getLcLevel(): LcLevel {
			return LcLevel.TETH
		}
	},
	HE {
		override fun getLcLevel(): LcLevel {
			return LcLevel.HE
		}
	},
	WAW {
		override fun getLcLevel(): LcLevel {
			return LcLevel.WAW
		}
	},
	ALEPH {
		override fun getLcLevel(): LcLevel {
			return LcLevel.ALEPH
		}
	}
}
